snuba/utils/streams/batching.py

```diff
@@ -190,75 +227,57 @@ def _run_once(self):
 
         self._handle_message(msg)
 
-    def signal_shutdown(self):
+    def signal_shutdown(self) -> None:
         """Tells the `BatchingKafkaConsumer` to shutdown on the next run loop iteration.
         Typically called from a signal handler."""
         logger.debug("Shutdown signalled")
 
         self.shutdown = True
 
-    def _handle_message(self, msg):
+    def _handle_message(self, msg: Message) -> None:
         start = time.time()
 
         # set the deadline only after the first message for this batch is seen
         if not self.__batch_deadline:
             self.__batch_deadline = self.max_batch_time / 1000.0 + start
 
-        try:
-            result = self.worker.process_message(msg)
-        except Exception:
-            if self.dead_letter_topic:
-                logger.exception("Error handling message, sending to dead letter topic.")
-                self.producer.produce(
-                    self.dead_letter_topic,
-                    key=msg.key(),
-                    value=msg.value(),
-                    headers={
-                        'partition': str(msg.partition()) if msg.partition() else None,
-                        'offset': str(msg.offset()) if msg.offset() else None,
-                        'topic': msg.topic(),
-                    },
-                    on_delivery=self._commit_message_delivery_callback,
-                )
-            else:
-                raise
+        result = self.worker.process_message(msg)
+        if result is not None:
+            self.__batch_results.append(result)
+
+        duration = (time.time() - start) * 1000
+        self.__batch_messages_processed_count += 1
+        self.__batch_processing_time_ms += duration
+        self.__metrics.timing("process_message", duration)
+
+        topic_partition_key = (msg.topic(), msg.partition())
+        if topic_partition_key in self.__batch_offsets:
+            self.__batch_offsets[topic_partition_key].hi = msg.offset()
         else:
-            if result is not None:
-                self.__batch_results.append(result)
-        finally:
-            duration = (time.time() - start) * 1000
-            self.__batch_messages_processed_count += 1
-            self.__batch_processing_time_ms += duration
-            self.__metrics.timing('process_message', duration)
-
-            topic_partition_key = (msg.topic(), msg.partition())
-            if topic_partition_key in self.__batch_offsets:
-                self.__batch_offsets[topic_partition_key][1] = msg.offset()
-            else:
-                self.__batch_offsets[topic_partition_key] = [msg.offset(), msg.offset()]
+            self.__batch_offsets[topic_partition_key] = Offsets(
+                msg.offset(), msg.offset()
+            )
```

Comment:
This looks scarier than it is — it's really just a big outdent due to the lack of dead letter queue.